lib/liquid/parse_context.rb

```diff
@@ -2,14 +2,15 @@
 
 module Liquid
   class ParseContext
-    attr_accessor :locale, :line_number, :trim_whitespace, :depth
+    attr_accessor :locale, :line_number, :trim_whitespace, :depth, :tags
     attr_reader :partial, :warnings, :error_mode
 
     def initialize(options = {})
       @template_options = options ? options.dup : {}
 
       @locale   = @template_options[:locale] ||= I18n.new
       @warnings = []
+      @tags = @template_options[:tags]
```

Comment:
Default to the global tags registry so it can be used unconditionally

```suggestion
      @tags = @template_options[:tags] || Liquid::Template.tags
```